### DISCUSSION TITLE
Apache config: Cookie http only / secure

### DISCUSSION THREAD

**Discussion by eglyn** [requarks/wiki#6866] [Error / Bug Report]
Hi :)

I tried to secure our apache server wich doing proxy for wikijs.
I added this line: `Header edit Set-Cookie ^(.*)$ $1;HttpOnly;Secure` to the virtualhost configuration.

But, now, I can't access to the admin console.
I can login to the wiki, but the administration button does not appear and when I mouseover my avatar, it says "connection" like I'm not connected :/

Can someone explain this ? 

Thx :)

**Comment by vahidirp** [comment]
It seems like the addition of the Header edit Set-Cookie ^(.*)$ $1;HttpOnly;Secure line in your Apache virtual host configuration might be causing an issue with the administration functionality of Wiki.js. This line is intended to add the HttpOnly and Secure flags to the Set-Cookie header for added security.

Here are a few suggestions to troubleshoot and resolve the issue:

    Check Apache Error Logs:
    Examine the Apache error logs for any error messages related to your Wiki.js application. Look for any specific error messages that might provide insights into the problem. The Apache error logs are usually located in /var/log/apache2/error.log or a similar directory.

    Review Wiki.js Logs:
    Check the logs generated by Wiki.js for any error messages or warnings. These logs are typically located in the directory where Wiki.js is installed. Look for any issues related to authentication or session handling.

    Verify the Set-Cookie Header Modification:
    Double-check the Header edit Set-Cookie ^(.*)$ $1;HttpOnly;Secure line in your Apache configuration. Ensure that it is correctly placed and does not inadvertently interfere with the functioning of Wiki.js. You may want to temporarily comment out this line and see if the issue persists.

    Browser Console:
    Open the browser console (press F12 or right-click and select "Inspect," then go to the "Console" tab) and check for any error messages or warnings when accessing the Wiki.js admin console. This may provide additional information about what is going wrong on the client side.

    Wiki.js Configuration:
    Review the Wiki.js configuration files to ensure that there are no settings that conflict with the HttpOnly and Secure flags on cookies. Look for any specific configuration options related to cookies or session management.

    Test Without the Header Modification:
    Temporarily remove the Header edit Set-Cookie ^(.*)$ $1;HttpOnly;Secure line and see if the issue persists. If the problem goes away, you can try alternative methods to secure the cookies or adjust the configuration to avoid conflicts.

    Consult Wiki.js Documentation:
    Check the official documentation of Wiki.js for any recommendations or best practices related to Apache configuration. There may be specific guidelines for integrating Wiki.js with a proxy server like Apache.

By systematically going through these steps, you should be able to identify the cause of the issue and find a solution. If the problem persists, you may also consider reaching out to the Wiki.js community or support channels for assistance.

**Comment by rrg92** [comment]
The current base code of Wiki.js does not support the use of HttpOnly cookies. 

This is because certain components of the Wiki interface rely on accessing cookies through JavaScript. Setting cookies to HttpOnly interferes with functionalities such as the display of the admin engineering icon, which indicates if a user has admin permissions.

This information, along with other permissions, is transmitted from the server to the client via cookies.
